kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/internal/asset/provider/AssetOptions.java

```diff
@@ -133,172 +93,111 @@ public AssetConfiguration getAssetConfiguration() {
-                final String cKey = key
-                        .substring(key.indexOf(DRIVER_SPECIFIC_PROPERTY_KEY) + DRIVER_SPECIFIC_PROPERTY_KEY.length());
-                channelConfig.put(cKey, value);
+            final String channelName = extractChannelName(key);
+            if (channelName == null) {
+                continue;
+            }
+            if (!targetChannelName.equals(channelName)) {
+                continue;
+            }
+            if (key.length() <= propertyBeginIndex) {
+                return null;
             }
+            channelConfig.put(key.substring(propertyBeginIndex), entry.getValue().toString());
         }
         return channelConfig;
     }
 
-    private String retrieveChannelName(final Map<String, Object> properties, final String channelKeyFormat) {
-        String result = null;
-        if (isNull(channelKeyFormat)) {
-            return result;
+    private String extractChannelName(String propertyKey) {
+        int pos = propertyKey.indexOf(CHANNEL_PROPERTY_SEPARATOR.value());
+        if (pos <= 0) {
+            return null;
         }
-
-        final String channelNamePropertyKey = channelKeyFormat + NAME.value();
-        if (properties.containsKey(channelNamePropertyKey)) {
-            result = (String) properties.get(channelNamePropertyKey);
-        }
-        return result;
+        return propertyKey.substring(0, pos);
     }
 
-    /**
-     * Retrieves the set of id of the channels from the properties
-     *
-     * @param properties
-     *            the properties to parse
-     * @return the list of channel IDs
-     * @throws NullPointerException
-     *             if the argument is null
-     */
-    private Set<Long> retrieveChannelIds(final Map<String, Object> properties) {
-        requireNonNull(properties, message.propertiesNonNull());
-        final Set<Long> channelIds = CollectionUtil.newHashSet();
-        for (final Map.Entry<String, Object> entry : properties.entrySet()) {
-            final String key = entry.getKey();
-            final List<String> strings = Arrays.asList(key.split(CHANNEL_PROPERTY_SPLIT_REGEX));
-            for (final String string : strings) {
-                if (string.matches("\\d+")) {
-                    channelIds.add(Long.parseLong(string));
-                }
+    private Map<String, Channel> retreiveChannelList(Map<String, Object> properties) {
+        Set<String> alreadyProcessedChannelNames = new HashSet<>();
+        Map<String, Channel> result = new HashMap<>();
+
+        for (Entry<String, Object> e : properties.entrySet()) {
+            String channelName = extractChannelName(e.getKey());
+            if (channelName == null || alreadyProcessedChannelNames.contains(channelName)) {
+                continue;
+            }
+            alreadyProcessedChannelNames.add(channelName);
+            if (!Channel.isValidChannelName(channelName)) {
```

Comment:
If you expect to encounter same channel names multiple times (irrespective of it to be valid or invalid), `contains` on set will definitely be optimal as it will be performed in constant time. 👍